### PR TITLE
Add ConnectionError to acceptable flaky exceptions for Freesound

### LINKED ISSUE #389: Freesound pull_data task fails when getting audio file size
## Description
The Freesound provider DAG is successfully pulling thousands of rows of data, but ultimately failing in its `pull_data` task with the following:

```
File "/usr/local/airflow/openverse_catalog/dags/providers/provider_api_scripts/freesound.py", line 238, in _get_audio_file_size
    return int(requests.head(url).headers["content-length"])
  File "/usr/local/airflow/.local/lib/python3.9/site-packages/requests/api.py", line 102, in head
    return request('head', url, **kwargs)
  File "/usr/local/airflow/.local/lib/python3.9/site-packages/requests/api.py", line 61, in request
    return session.request(method=method, url=url, **kwargs)
  File "/usr/local/airflow/.local/lib/python3.9/site-packages/requests/sessions.py", line 529, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/airflow/.local/lib/python3.9/site-packages/requests/sessions.py", line 645, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/airflow/.local/lib/python3.9/site-packages/requests/adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='freesound.org', port=443): Max retries exceeded with url: /data/previews/289/289097_2050105-hq.mp3
```

## Reproduction

Attempt to run the Freesound DAG in Airflow and check the logs for the pull_audio_data task.

## Additional context

Noting that the `_get_audio_file_size` function contains the following warning: ["Freesound can be a bit finicky, so we want to retry it a few times"](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/providers/provider_api_scripts/freesound.py#L235) as well as a TODO to move filesize detection to the polite crawler. This would be a good place to start investigating.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #389 by @stacimc

## Description

Similar to #330, this PR adds `ConnectionError` to the list of potentially flaky exceptions from Freesound that should be retried. These exceptions occur when attempting to get the file size for the previews, and aren't strictly necessary data to have. We've found that retrying with a delay fixes most of these occurrences. In cases where it does not, we exclude the file size information and continue iterating through the API.

I did a bit of refactoring to make this easier for us to change if we needed to down the line. Previously there were 3 locations we had to update to add a new exception, now there's only 1!

## Testing Instructions

`just test`! (I love that that's a full sentence too :grin:)

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by stacimc — APPROVED** [review]
Looks great, and I appreciate the refactor 💯

**Review by krysal — APPROVED** [review]
LGTM! So concise and well handled ✨

File: tests/dags/providers/provider_api_scripts/test_freesound.py [hunk 1]
```diff
@@ -54,11 +53,20 @@ def test_get_audio_pages_returns_correctly_with_no_results():
...
 
 
-def test_get_audio_file_size_retries_and_does_not_raise(audio_data):
+@pytest.mark.parametrize(
+    "exception_type",
+    [
+        # These are fine
+        *freesound.FLAKY_EXCEPTIONS,
+        # This should raise immediately
+        pytest.param(ValueError, marks=pytest.mark.raises(exception=ValueError)),
```
[the remark this answers is not in the document]
Comment (AetherUnbound) [reply tests/dags/providers/provider_api_scripts/test_freesound.py:62]: Yea! The `raises` mark isn't something that comes baked into pytest but fortunately [there's a plugin for it that we use!](https://github.com/Lemmons/pytest-raises)

[the remark this answers is not in the document]
Comment (AetherUnbound) [reply tests/dags/providers/provider_api_scripts/test_freesound.py:62]: It used to be even easier/with less boilerplate but unfortunately [that was deprecated](https://github.com/pytest-dev/pytest/issues/4395#issuecomment-438839270) :sob:

**Pull request merged**